pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -0,0 +1,385 @@
+package servicebindingrequest
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	osappsv1 "github.com/openshift/api/apps/v1"
+	olmv1alpha1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	extv1beta1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	v1alpha1 "github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+)
+
+const (
+	connectsToLabel                = "connects-to"
+	almDescriptorPrefix            = "urn:alm:descriptor"
+	almKubernetesPrefix            = "io.kubernetes"
+	almServiceBindingRequestPrefix = "io.servicebindingrequest"
+)
+
+// Reconciler reconciles a ServiceBindingRequest object
+type Reconciler struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// decodeString encoded with base64.
+func decodeString(encoded string) (string, error) {
+	decoded, err := base64.StdEncoding.DecodeString(encoded)
+	if err != nil {
+		return "", err
+	}
+	return string(decoded), nil
+}
+
+// pathValue read value of a given "path" inside of informed custom resource definition instance.
+func (r *Reconciler) pathValue(crd *unstructured.Unstructured, path string) (string, error) {
+	object := crd.Object
+	status, exists := object["status"].(map[string]interface{})
+	if !exists {
+		return "", fmt.Errorf("unable to find 'status' field in object '%s'", crd.GetName())
+	}
+
+	value, exists := status[path]
+	if !exists {
+		return "", fmt.Errorf("unable to find attribute for path '%s'", path)
+	}
+
+	return fmt.Sprintf("%v", value), nil
+}
+
+// parseXDescriptor parse's OLM CustomResourceDefinition descripton in order to return it's kind
+// and attribute.
+func (r *Reconciler) parseXDescriptor(xDescriptor string) (string, string) {
+	if !strings.HasPrefix(xDescriptor, almDescriptorPrefix) {
+		return "", ""
+	}
+
+	prefix := fmt.Sprintf("%s:%s", almDescriptorPrefix, almServiceBindingRequestPrefix)
+	if !strings.HasPrefix(xDescriptor, prefix) {
+		return "", ""
+	}
+
+	sections := strings.Split(xDescriptor, ":")
+	kind := sections[len(sections)-2]
+	attribute := sections[len(sections)-1]
+
+	return kind, attribute
+}
+
+// generateFieldName based on the x-descriptor value, generating a string with the same standards
+// than environment variable naming.
+func (r *Reconciler) generateFieldName(xDescriptor string) string {
```

Comment:
Do we really need the complete Xdescriptor or can we take the last segment after the ":". For eg- <CR NAME>_<keyvalue>

Reply:
For eg if CR is `postgres` and `key` is password then the env variable will become `POSTGRES_PASSWORD`

Reply:
@Avni-Sharma that's a good question, I think @sbose78 and @baijum can share more information.